python/tests/integration/connectors/memory/test_postgres.py

```diff
@@ -1,214 +1,206 @@
+        },
+        container_mode=True,
+        to_dict=lambda x: x.to_dict(orient="records"),
+        from_dict=lambda x, **_: pd.DataFrame(x),
+    )
+    df = pd.DataFrame([record])
+    return definition, df
+
+
+@pytest_asyncio.fixture(scope="function")
+async def vector_store() -> AsyncGenerator[PostgresStore, None]:
+    async with await pg_settings.create_connection_pool() as pool:
+        yield PostgresStore(connection_pool=pool)
+
+
+@pytest_asyncio.fixture(scope="function")
+async def simple_collection(vector_store: PostgresStore):
+    suffix = str(uuid.uuid4()).replace("-", "")[:8]
+    collection_id = f"test_collection_{suffix}"
+    collection = vector_store.get_collection(collection_id, SimpleDataModel)
+    await collection.create_collection()
+    yield collection
+    await collection.delete_collection()
+
+
+def test_create_store(vector_store):
+    assert vector_store.connection_pool is not None
 
 
 @pytest.mark.asyncio
-async def test_create_and_does_collection_exist(connection_string):
-    memory = PostgresMemoryStore(connection_string, 2, 1, 5)
-    try:
-        await memory.create_collection("test_collection")
-        result = await memory.does_collection_exist("test_collection")
-        assert result is not None
-    except PoolTimeout:
-        pytest.skip("PoolTimeout exception raised, skipping test.")
+async def test_create_does_collection_exist_and_delete(vector_store: PostgresStore):
+    suffix = str(uuid.uuid4()).replace("-", "")[:8]
+    collection = vector_store.get_collection(f"test_collection_{suffix}", SimpleDataModel)
 
+    does_exist_1 = await collection.does_collection_exist()
+    assert does_exist_1 is False
 
-@pytest.mark.asyncio
-async def test_get_collections(connection_string):
-    memory = PostgresMemoryStore(connection_string, 2, 1, 5)
+    await collection.create_collection()
+    does_exist_2 = await collection.does_collection_exist()
+    assert does_exist_2 is True
 
-    try:
-        await memory.create_collection("test_collection")
-        result = await memory.get_collections()
-        assert "test_collection" in result
-    except PoolTimeout:
-        pytest.skip("PoolTimeout exception raised, skipping test.")
+    await collection.delete_collection()
+    does_exist_3 = await collection.does_collection_exist()
+    assert does_exist_3 is False
 
 
 @pytest.mark.asyncio
-async def test_delete_collection(connection_string):
-    memory = PostgresMemoryStore(connection_string, 2, 1, 5)
-    try:
-        await memory.create_collection("test_collection")
-
-        result = await memory.get_collections()
-        assert "test_collection" in result
-
-        await memory.delete_collection("test_collection")
-        result = await memory.get_collections()
-        assert "test_collection" not in result
-    except PoolTimeout:
-        pytest.skip("PoolTimeout exception raised, skipping test.")
+async def test_list_collection_names(vector_store, simple_collection):
```

Comment:
Question: will the `vector_store` fixture return the same store instance as the onw that `simple_collection` fixture requests?

Reply:
It will; both `async def test_list_collection_name` and `async def simple_collection` use the `vector_store` feature, which is only created once per scope (in this case, the test function call). So the fixture will produce the same instance any time it's invoked during the test function call.